### PR TITLE
planner: fix not check children size > 0 in late materialization

### LINKED ISSUE #42384: TiDB panic for Tiflash late materilation set children
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```


[2023/03/20 10:35:04.284 +08:00] [ERROR] [conn.go:1039] ["connection running loop panic"] [conn=4362220024224640871] [lastSQL="explain SELECT POWER((d_next_o_id -1 - mo), 2) + POWER((d_next_o_id -1 - mno), 2) diff FROM district dis, (SELECT o_d_id,max(o_id) mo FROM orders WHERE o_w_id= 1 GROUP BY o_d_id) q, (select no_d_id,max(no_o_id) mno from new_order where no_w_id= 1 group by no_d_id) no where d_w_id = 1 and q.o_d_id=dis.d_id and no.no_d_id=dis.d_id"] [err="runtime error: index out of range [0] with length 0"] [stack="github.com/pingcap/tidb/server.(*clientConn).Run.func1
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1042
runtime.gopanic
\t/usr/local/go/src/runtime/panic.go:884
github.com/pingcap/tidb/executor.(*Compiler).Compile.func1
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/compiler.go:54
runtime.gopanic
\t/usr/local/go/src/runtime/panic.go:884
runtime.goPanicIndex
\t/usr/local/go/src/runtime/panic.go:113
github.com/pingcap/tidb/planner/core.(*PhysicalIndexReader).SetChildren
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/physical_plans.go:352
github.com/pingcap/tidb/planner/core.predicatePushDownToTableScan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/tiflash_selection_late_materialization.go:76
github.com/pingcap/tidb/planner/core.predicatePushDownToTableScan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/tiflash_selection_late_materialization.go:74
github.com/pingcap/tidb/planner/core.predicatePushDownToTableScan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/tiflash_selection_late_materialization.go:74
github.com/pingcap/tidb/planner/core.predicatePushDownToTableScan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/tiflash_selection_late_materialization.go:74
github.com/pingcap/tidb/planner/core.predicatePushDownToTableScan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/tiflash_selection_late_materialization.go:74
github.com/pingcap/tidb/planner/core.predicatePushDownToTableScan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/tiflash_selection_late_materialization.go:74
github.com/pingcap/tidb/planner/core.tryEnableLateMaterialization
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/optimizer.go:557
github.com/pingcap/tidb/planner/core.postOptimize
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/optimizer.go:403
github.com/pingcap/tidb/planner/core.DoOptimize
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/optimizer.go:307
github.com/pingcap/tidb/planner.optimize
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:462
github.com/pingcap/tidb/planner.Optimize
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:290
github.com/pingcap/tidb/planner/core.(*PlanBuilder).buildExplain
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/planbuilder.go:4971
github.com/pingcap/tidb/planner/core.(*PlanBuilder).Build
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/planbuilder.go:794
github.com/pingcap/tidb/planner.buildLogicalPlan
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:506
github.com/pingcap/tidb/planner.optimize
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:427
github.com/pingcap/tidb/planner.Optimize
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:290
github.com/pingcap/tidb/executor.(*Compiler).Compile
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/compiler.go:98
github.com/pingcap/tidb/session.(*session).ExecuteStmt
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/session/session.go:2140
github.com/pingcap/tidb/server.(*TiDBContext).ExecuteStmt
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/driver_tidb.go:259
github.com/pingcap/tidb/server.(*clientConn).handleStmt
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2001
github.com/pingcap/tidb/server.(*clientConn).handleQuery
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1851
github.com/pingcap/tidb/server.(*clientConn).dispatch
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1337
github.com/pingcap/tidb/server.(*clientConn).Run
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1120
github.com/pingcap/tidb/server.(*Server).onConn
\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/server.go:675"]
```
### 4. What is your TiDB version? (Required)

| Release Version: v6.7.0-alpha
Edition: Community
Git Commit Hash: 56412f514ecad4209cb0d92878b68644ed94469a
Git Branch: heads/refs/tags/v6.7.0-alpha
UTC Build Time: 2023-03-18 11:42:18
GoVersion: go1.20.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv |

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: close #42384

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/tiflash_selection_late_materialization.go [hunk 1]
```diff
@@ -69,11 +69,13 @@ func predicatePushDownToTableScan(sctx sessionctx.Context, plan PhysicalPlan) Ph
 	case *PhysicalTableReader:
 		predicatePushDownToTableScan(sctx, p.tablePlan)
```
Comment (hawkingrei) [inline planner/core/tiflash_selection_late_materialization.go:70]: Can you add tests for it?

**Review by breezewish — APPROVED** [review]

File: planner/core/tiflash_selection_late_materialization.go (hunk 1 above)
Comment (Lloyd-Pottiger) [reply planner/core/tiflash_selection_late_materialization.go]: have added before, due to didnot merge  master.

Comment (Lloyd-Pottiger) [reply planner/core/tiflash_selection_late_materialization.go]: https://github.com/pingcap/tidb/pull/42349

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**